src/java/org/apache/cassandra/db/memtable/AbstractAllocatorMemtable.java

```diff
@@ -78,9 +79,17 @@ public abstract class AbstractAllocatorMemtable extends AbstractMemtableWithComm
     static MemtablePool createMemtableAllocatorPool()
     {
         Config.MemtableAllocationType allocationType = DatabaseDescriptor.getMemtableAllocationType();
-        long heapLimit = DatabaseDescriptor.getMemtableHeapSpaceInMiB() << 20;
-        long offHeapLimit = DatabaseDescriptor.getMemtableOffheapSpaceInMiB() << 20;
-        float memtableCleanupThreshold = DatabaseDescriptor.getMemtableCleanupThreshold();
+
+        long heapLimit = new DataStorageSpec.IntMebibytesBound((int) (Runtime.getRuntime().maxMemory() / (4 * 1048576))).toMebibytes() << 20;
```

Comment:
I took this from the logic when `DatabaseDescriptor` is normally initialized in daemon mode.